client.go

```diff
@@ -22,8 +22,25 @@ package cluster
 
 import (
 	"github.com/m3db/m3cluster/kv"
+	"github.com/m3db/m3x/watch"
 )
 
+// A Shard represents a piece of data owned by the service
+type Shard interface {
+	ID() uint32 // ID returns the id of the shard
+}
+
+// NewShard returns a new Shard
+func NewShard(s uint32) Shard { return &shard{id: s} }
```

Comment:
nit: don't need to return a pointer here. The id is immutable after the shard is created, so can just be a value and avoid a heap allocation.